### ISSUE TITLE
Training YOLOv3-Tiny-PRN

### ISSUE THREAD

**Issue by VideoPlayerCode** [WongKinYiu/PartialResidualNetworks#7]
Hi!

The PRN models has more layers (shortcut layers). How many layers should we extract from weights when training?

Still 15 layers, like for regular YOLOV3-Tiny? https://github.com/AlexeyAB/darknet#how-to-train-tiny-yolo-to-detect-your-custom-objects

**Comment by WongKinYiu** [comment]
yes, still 15, shortcut layer do not have weights.
if u want to using `mask = 0,1,2`, i think re-train coco with same setting for transfer learning will be slightly better.

however, for coco dataset, i use same imagenet pre-trained model, `mask = 1,2,3` get better than `mask = 0,1,2`.

**Comment by VideoPlayerCode** [comment]
> yes, still 15, shortcut layer do not have weights.

Oh okay, that makes sense. Thank you! I've trained the data now and presented the results! <3 https://github.com/opencv/opencv/pull/15739#issuecomment-544931445

> if u want to using `mask = 0,1,2`, i think re-train coco with same setting for transfer learning will be slightly better.

Yes probably better.

> however, for coco dataset, i use same imagenet pre-trained model, `mask = 1,2,3` get better than `mask = 0,1,2`.

Have you trained COCO `0,1,2` _and_ `1,2,3` weights? Or just `1,2,3` weights?

If you have _trained_ COCO weights on both values, I'd love to hear about what you did and what the results were.

**Comment by WongKinYiu** [comment]
yes, i have trained both of `0,1,2` and `1,2,3` on coco.
codalab was dead few month ago, so i lost the results of those models.

**Comment by VideoPlayerCode** [comment]
Accuracy and speed results:

https://github.com/opencv/opencv/pull/15739#issuecomment-544931445
https://github.com/AlexeyAB/darknet/issues/4091#issuecomment-542513900

Hint for everyone: This PRN network needs more iterations than regular non-PRN versions of YOLO to learn good accuracy. Train train train! If you don't get good mAP, check your amount of training iterations.

**Comment by NgTuong** [comment]
@WongKinYiu ,
Can you share pretrained weight  of YOLOv3-tiny-PRN?
this weight Link not have!

**Comment by WongKinYiu** [comment]
@NgTuong hello,

u can follow https://github.com/AlexeyAB/darknet#how-to-train-tiny-yolo-to-detect-your-custom-objects to get coco pre-trained model.
i m in iccv now, so if u want to use imagenet pre-trained model, i will put it on the github after iccv.

**Comment by NgTuong** [comment]
@WongKinYiu thank you but i have a problem when training
![Screenshot_2019-10-29 Google Colaboratory](https://user-images.githubusercontent.com/42471216/67731179-edb25000-fa29-11e9-84bc-a092c45212ea.png).

**Comment by WongKinYiu** [comment]
i think u r missing --clear flag.

by the way, using yolov3-tiny.weights as pre-trained model is not the best solution for transfer learning.

**Comment by NgTuong** [comment]
@WongKinYiu,
In this case, what do i?
Not use pretrained?

**Comment by VideoPlayerCode** [comment]
> In this case, what do i?

https://github.com/WongKinYiu/PartialResidualNetworks/tree/master/model

**Comment by WongKinYiu** [comment]
@VideoPlayerCode  thanks.

**Comment by NgTuong** [comment]
@VideoPlayerCode , @WongKinYiu ,
thanks =)

**Comment by NgTuong** [comment]
hi @WongKinYiu , @VideoPlayerCode .
![Screenshot_2019-10-30 Google Colaboratory](https://user-images.githubusercontent.com/42471216/67841048-cb076080-fb29-11e9-9829-13bcfdf02d9f.png)
it don't train too?

**Comment by NgTuong** [comment]
@WongKinYiu ,
why must use --clear flag?

**Comment by WongKinYiu** [comment]
@NgTuong 

becuz there are some logs in the weight file.
for example, trained iterations.
pretrained model recorded the max iteration, for yolov3 case, it is 500200.
however u want to train ur model starts from 0, so u need to use --clear flag.

**Comment by NickiBD** [comment]
Hi @WongKinYiu

I trained yolo-tiny-prn.cfg  with yolo-tiny-conv.15 for COCO  dataset as pretrain model.
I also got a lower mAp for yolov3-tiny-prn.cfg As you have mentioned ,it is due to using yolo-tiny-conv.15  instead of prn.conv.15. Could you please tell me how you get the prn.conv.15 from the weight you provided ?Many thanks .
